core/iwasm/fast-jit/jit_frontend.c

```diff
@@ -487,13 +488,50 @@ get_ent_and_bw_cnts_reg(JitFrame *frame)
              * ent_and_br_cnts = (uint32*)mod_ent_and_br_cnts +
              *     cur_wasm_func_idx
              */
-            GEN_INSN(LDPTR, frame->ent_and_bw_cnts_reg, mod_ent_and_bw_cnts_reg,
+            GEN_INSN(LDPTR, frame->ent_and_br_cnts_reg, mod_ent_and_bw_cnts_reg,
                      offset_reg);
         }
     }
 
-    return frame->ent_and_bw_cnts_reg;
+    return frame->ent_and_br_cnts_reg;
+}
+
+void
+gen_increase_cnt_insn(JitFrame *frame, uint32 cnt_idx)
+{
+    JitCompContext *cc = frame->cc;
+    JitReg ent_and_br_cnts_reg = get_ent_and_br_cnts_reg(frame);
+    JitReg cnt_reg = jit_cc_new_reg_I32(cc);
+    JitReg cnt_inc_reg = jit_cc_new_reg_I32(cc);
+    JitReg offset_reg;
+
+    bh_assert(cnt_idx > 0);
+
+    if (UINTPTR_MAX == UINT64_MAX) {
+        offset_reg = jit_cc_new_reg_I64(cc);
+        /* uint32[cnt_idx] -> x bytes*/
+        GEN_INSN(SHL, offset_reg, NEW_CONST(I64, cnt_idx), NEW_CONST(I64, 2));
+    }
+    else {
+        offset_reg = jit_cc_new_reg_I32(cc);
+        /* uint32[cnt_idx] -> x bytes*/
+        GEN_INSN(SHL, offset_reg, NEW_CONST(I32, cnt_idx), NEW_CONST(I32, 2));
+    }
+
+    /* load -> increase -> store */
+    JitInsn *insn = GEN_INSN(LDI32, cnt_reg, ent_and_br_cnts_reg, offset_reg);
```

Comment:
insn isn't used later, had better remove it?

Reply:
`insn` belongs to another PR. Let's keep it here. It will be used.